tach/logging/logger.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+import logging
+import os
+import threading
+from typing import Any, Optional
+
+from pydantic import BaseModel, Field
+
+from tach import cache
+from tach.logging.api import log_record, log_uid
+from tach.parsing import parse_project_config
+
+
+class LogDataModel(BaseModel):
+    function: str
+    parameters: dict[str, Any] = Field(default_factory=dict)
+
+
+def send_log_entry(record: logging.LogRecord, entry: str) -> None:
+    is_ci = "CI" in os.environ
+    data: Optional[LogDataModel] = getattr(record, "data", None)
+    uid = cache.get_uid()
+    log_data: dict[str, Any] = {
+        "user": str(uid),
+        "message": entry,
+        "level": record.levelname,
+        "timestamp": record.created,
+        "function": data.function if data else None,
+        "parameters": data.parameters if data else None,
+    }
+    log_uid(uid, is_ci)
+    log_record(log_data)
+
+
+class RemoteLoggingHandler(logging.Handler):
+    def emit(self, record: logging.LogRecord) -> None:
+        log_entry = self.format(record)
+        thread = threading.Thread(target=send_log_entry, args=(record, log_entry))
+        thread.start()
```

Comment:
I think this should be a daemon thread so that it can't cause the process to hang.

https://docs.python.org/3/library/threading.html#threading.Thread.daemon